### PR TITLE
Libva for ffmpeg encoding in intel hardware

### PR DESCRIPTION
**Pull request by hmaarrfk**
This PR packages

  - libva
  - libva-utils
  - libpciaccess
  - libdrm
  - intel-vaapi-driver
  - intel-hybrid-driver
  - cmrt

These are all necessary to get intel hardware encoding to work with FFMPEG

### CODE REVIEW

**Comment by hmaarrfk** [comment]
@pkgw have you been able to compile with xorg-macros?

**Comment by pkgw** [comment]
@hmaarrfk Do you need to pass `-I` flags to autoreconf? See [this link](https://github.com/conda-forge/xorg-libx11-feedstock/blob/master/recipe/build.sh#L31-L40) for how this is typically done in the X.org packages. (Note that these jump through a few extra hoops since they route the Windows builds to build.sh.)

**Comment by hmaarrfk** [comment]
@pkgw thanks for the pointer. I just looked at my prefix and xorg-macros seems to be the only thing installed in `share/pkgconfig`, everthing else is in `lib/pkgconfig` i wonder if that is the source of the issue

**Comment by pkgw** [comment]
@hmaarrfk `aclocal`, not `pkgconfig`?

**Comment by hmaarrfk** [comment]
right, but pkgconfig tells the system where the headers are typically. It should be able to auto detect without me having to specify an extra include directory.

**Comment by pkgw** [comment]
@hmaarrfk The `autoreconf` stage happens before that. The `-I` flag here is a search path for Autoconf macros, not C header files. There are special X.org macros that are used when processing `configure.ac` to create `configure`, which is one of the things that `autoreconf` does. The `configure` script then invokes pkg-config to get information about where to find C headers.

**Comment by hmaarrfk** [comment]
@pkgw thanks for the explanation.

I had to add `export ACLOCAL="aclocal -I ${PREFIX}/share/aclocal"` for things to work. Passing the `-I` to autoreconf didn't work.

```
set -ex
export ACLOCAL="aclocal -I ${PREFIX}/share/aclocal"
autoreconf --install
./autogen.sh --prefix=$PREFIX
make install
```
is what I have now, lets see what the CI's say.

**Comment by pkgw** [comment]
Yes, that's the standard practice. It's not ideal but it's the usual approach. [Example](https://github.com/conda-forge/fontconfig-feedstock/blob/master/recipe/meta.yaml#L64-L82).

**Comment by hmaarrfk** [comment]
Thanks so much @pkgw I couldn't have done this without your help!!

**Comment by pkgw** [comment]
Happy to help!

**Comment by hmaarrfk** [comment]
Its actually quite cool. With this an the FFMPEG PR, I'm able to encode 4k(ish) video using minimal CPU.

Not to mention, all these libraries are MIT/BSD, so one can get rid of the GPL dependency on x264 if they choose.

File: recipes/cmrt/build.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# https://01.org/linuxgraphics/documentation/build-guide-0
+
+export CFLAGS="${CFLAGS} -lrt"
+export CXXFLAGS="${CXXFLAGS} -lrt"
+./autogen.sh --prefix=$PREFIX
+make -j$(nproc) install
```
Comment (ocefpaf) [inline recipes/cmrt/build.sh]: No make check?

File: recipes/cmrt/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    - test -f $PREFIX/include/cm_rt_linux.h
+
+about:
+  home: https://github.com/intel/cmrt
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: 'Media GPU kernel manager for Intel G45 & HD Graphics family'
+
+  # The remaining entries in this section are optional, but recommended
```
Comment (ocefpaf) [inline recipes/cmrt/meta.yaml]: Can you remove the commend and reduce the description a little bit?

**Comment by ocefpaf** [comment]
Two comments that can be addressed in the feestock.

**Comment by hmaarrfk** [comment]
Thanks!

File: recipes/cmrt/build.sh (hunk 1 above)
Comment (hmaarrfk) [reply recipes/cmrt/build.sh]: should I use `make check` for most packages?

**Comment by sodre** [comment]
I know this already got merged, but shouldn't we pin the dependencies a little bit more? e.g. libva <version>.* ?

**Comment by hmaarrfk** [comment]
@sodre, so for every c library dependency, i should pin it in the `run` section as `libva .*`? 
Will the bot trigger a chain of upgrades when necessary?

I mostly packaged python dependencies before so I only used the `>=` constructs.

**Comment by sodre** [comment]
@hmaarrfk, it might be work reading the following sections:
 - How to export runtime requirements: https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#export-runtime-requirements
 - Package pinning: https://conda-forge.org/docs/guidelines.html#fixing-broken-packages

**Comment by hmaarrfk** [comment]
@sodre, so it seems I need to add a 
```
build:
  run_exports:
    - {{ pin_subpackage('package_name', max_pin='x.x') }}
```
to most of my packages.

I think that luckily, most packages didn't build due to dependencies not being available.

**Comment by sodre** [comment]
I would start with the base package -- in your case it seems to be libdrm -- and work towards intel-vaapi-driver. As you add run_exports to each one of the libraries, you should be able to remove them from the requirements.run on each of downstream packages.

Feel free to ping me in the feedstock that you want me to review.

**Comment by hmaarrfk** [comment]
Thanks @sodre that was what I what I understood from those docs.

**Pull request merged**